src/panels/KaitoModelsPanel.ts

```diff
@@ -166,35 +292,12 @@ export class KaitoModelsPanelDataProvider implements PanelDataProvider<"kaitoMod
             kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
             if (failed(kubectlresult)) {
                 vscode.window.showErrorMessage(kubectlresult.error);
-                return {
-                    clusterName: this.clusterName,
-                    modelName: "",
-                    workspaceExists: false,
-                    resourceReady: null,
-                    inferenceReady: null,
-                    workspaceReady: null,
-                    age: 0,
-                } as InitialState;
+                return this.getInitialState();
```

Comment:
Suggestion - This can be done as separate PR @tejhan where you can add telemetry events to track failures/success which will help drive more enhancements/bug fixes for this feature.

Reply:
Makes sense, will look into including this for next PR